tests/framework/e2e/etcd_process.go

```diff
@@ -253,6 +259,34 @@ func (ep *EtcdServerProcess) Close() error {
 	return nil
 }
 
+func (ep *EtcdServerProcess) VerifySchemaVersion(lg *zap.Logger) error {
+	dbPath := datadir.ToBackendFileName(ep.cfg.DataDirPath)
+	be := backend.NewDefaultBackend(lg, dbPath)
+	defer be.Close()
+	ver, err := schema.UnsafeDetectSchemaVersion(lg, be.BatchTx())
+	if err != nil {
+		return err
+	}
+	maxStorageVersion := version.V3_6
+	if ep.cfg.ExecPath == BinPath.EtcdLastRelease {
+		maxStorageVersion = version.V3_5
+	}
+	// in a mix version cluster, the storage version would be set to the cluster version,
+	// which could be lower than the server version by 1 minor version.
+	if !ver.LessThan(maxStorageVersion) && !ver.Equal(maxStorageVersion) {
+		return fmt.Errorf("expect backend schema version to be <= %s, but got %s", maxStorageVersion.String(), ver.String())
+	}
+	// check new fields introduced in V3_6 do not exist in V3_5 data file.
+	// V3_6 contains all the fields in V3_5, so no need to check for V3_6 servers.
+	if maxStorageVersion == version.V3_5 {
```

Comment:
ditto, the generic rule to verify: each storageVersion shouldn't include any fields which are only supposed to be in higher version.

Reply:
to generalize the rule, need to use `schemaChanges` to check the changes. So I added some more complex methods to check that.

Reply:
The solution does work, but it sacrifices readability and lacks elegance to me. Essentially you just need to get a list of new fields in the higher version in this case. Please read https://github.com/etcd-io/etcd/pull/17881#issuecomment-2084889021